proposals/20240901-drop-support-for-syscall-enter-events.md

```diff
@@ -0,0 +1,353 @@
+  * **The `sys_enter` tracepoint takes half of the time kernel side.** This is in some way expected, `sys_enter` and `sys_exit` dispatchers do almost the same thing.
+* As we can see from Redis Data, we can handle 7000/8000 req/s less than in the "ideal" case (with no instrumentation).
+
+### Small syscall set
+
+We examine the overhead imposed by tracing two system calls frequently called by our redis-benchmark: `sendto`, `recvfrom`.
+
+```text
+----------------------------- STATS ------------------------------
+
+Time elapsed: 227 s
+
+------------> Kernel stats
+Seen by driver (kernel side events): 44.000.008
+Rate of kernel side events (events/second): 193.832
+Stats v2: 27 metrics in total
+[1] kernel-side counters
+[2] libbpf stats (compare to `bpftool prog show` CLI)
+[1] n_evts: 44.000.008
+[2] sys_enter.run_cnt: 235.481.342
+[2] sys_enter.run_time_ns: 27.009.550.840
+[2] sys_enter.avg_time_ns: 114
+[2] sys_exit.run_cnt: 235.481.390
+[2] sys_exit.run_time_ns: 27.793.327.491
+[2] sys_exit.avg_time_ns: 118
+[2] total_run_cnt: 470.962.732
+[2] total_run_time_ns: 54.802.878.331
+[2] total_avg_time_ns: 116
+
+------------> Userspace stats
+Number of `SCAP_SUCCESS` (events correctly captured): 44.000.008
+Number of `SCAP_TIMEOUTS`: 336.373
+Number of `scap_next` calls: 44.336.381
+Number of bytes received: 2.555.000.598 bytes (2,8 GB)
+Average dimension of events: 58 bytes
+Rate of userspace events (events/second): 193.832
+Syscall stats (userspace-side):
+- [sendto__enter]: 11.000.003
+- [sendto__exit]: 11.000.003
+- [recvfrom__enter]: 11.000.001
+- [recvfrom__exit]: 11.000.001
+
+
+------------------------------------------------------------------
+
+PING_INLINE: 48873.46 requests per second, p50=0.527 msec                
+PING_MBULK: 48510.72 requests per second, p50=0.527 msec                
+SET: 48435.53 requests per second, p50=0.527 msec                
+GET: 48685.49 requests per second, p50=0.527 msec                
+INCR: 48206.71 requests per second, p50=0.527 msec                
+LPUSH: 48407.39 requests per second, p50=0.527 msec                
+RPUSH: 48306.84 requests per second, p50=0.527 msec                
+LPOP: 48440.23 requests per second, p50=0.527 msec                
+RPOP: 48600.31 requests per second, p50=0.527 msec                
+SADD: 48612.12 requests per second, p50=0.527 msec                
+HSET: 47684.90 requests per second, p50=0.535 msec
+```
+
+* Our ebpf programs run `54` s in the kernel. **`~27 s` in the enter fillers and `~27 s` in the exit one.**
+* We see a slowdown both in `scap-open` and in `redis`.
+  * scap-open now takes `227 s` instead of `194 s`.
+  * we have a decrease of `7000/8000` requests/sec in Redis.
+* We send to userspace `2.8` GB in `227` s. Without enter events we should send almost the half, even if enter events are usually smaller than the exit ones, so it's not completely true.
```

Comment:
Why would you send half of the data? The proposal is to move all data to exit events, that means you'll only be saving the size of the enter headers which, I would assume are a very small portion of the actual event size.

Reply:
removed :)